pkg/appclient/relay_verifier.go

```diff
@@ -0,0 +1,47 @@
+package appclient
+
+import (
+	"context"
+
+	"github.com/cometbft/cometbft/crypto"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"pocket/x/service/types"
+)
+
+// verifyResponse verifies the relay response signature.
+func (app *appClient) verifyResponse(
+	ctx context.Context,
+	supplierAddress string,
+	relayResponse *types.RelayResponse,
+) error {
+	// Query for the supplier account to get the application's public key to verify the relay request signature.
+	accQueryReq := &accounttypes.QueryAccountRequest{Address: supplierAddress}
+	accQueryRes, err := app.accountQuerier.Account(ctx, accQueryReq)
```

Comment:
Here we may want to maintain a cache of the node's pubkeys (those in the session) against their addresses, or include the pubkey with the signature (the node would just need to store there own). The idea is that this would reduce latency and reduce the number of on chain queries we have to do, instead of each verification only just when cache is missing.

Consider adding a `TODO` here